web/src/beta/lib/core/engines/Cesium/Sketch/index.tsx

```diff
@@ -0,0 +1,108 @@
+import { Color } from "@cesium/engine";
+import { Cartesian3 } from "cesium";
+import { type LineString, type MultiPolygon, type Polygon } from "geojson";
+import { memo, useMemo, type FC } from "react";
+import { type RequireExactlyOne } from "type-fest";
+
+import { SketchType } from "../../../Map/Sketch/types";
+import { Position3d } from "../../../types";
+
+import { createGeometry, GeometryOptions } from "./createGeometry";
+import ExtrudedControlPoints from "./ExtrudedControlPoints";
+import { ExtrudedPolygonEntity } from "./ExtrudedPolygonEntity";
+import { PolygonEntity } from "./PolygonEntity";
+import { PolylineEntity } from "./PolylineEntity";
+import SurfaceControlPoints from "./SurfaceControlPoints";
+import { convertGeometryToPositionsArray, convertPolygonToHierarchyArray } from "./utils";
+
+export type SketchComponentProps = RequireExactlyOne<
+  {
+    geometry?: LineString | Polygon | MultiPolygon | null;
+    geometryOptions?: {
+      type: SketchType;
+      controlPoints: readonly Position3d[];
+    } | null;
+    extrudedHeight?: number;
+    disableShadow?: boolean;
+    color?: string;
+  },
+  "geometry" | "geometryOptions"
+>;
+
+const DEFAULT_SKETCH_COLOR = "#00bebe";
+
+const SketchComponent: FC<SketchComponentProps> = memo(
+  ({ geometry, geometryOptions, extrudedHeight, disableShadow, color: stringColor }) => {
+    const cartesianGeometryOptions: GeometryOptions | null = useMemo(
+      () =>
+        geometryOptions
+          ? {
+              ...geometryOptions,
+              controlPoints: geometryOptions?.controlPoints.map(p => new Cartesian3(...p)),
+            }
+          : null,
+      [geometryOptions],
+    );
+
+    const g = useMemo(
+      () =>
+        geometry ?? (cartesianGeometryOptions ? createGeometry(cartesianGeometryOptions) : null),
+      [geometry, cartesianGeometryOptions],
+    );
+
+    const { positionsArray, hierarchyArray } = useMemo(() => {
+      if (g?.type === "Point") {
+        return {};
+      } else if (g?.type === "LineString") {
+        return { positionsArray: convertGeometryToPositionsArray(g) };
+      } else if (g != null) {
+        return {
+          positionsArray: convertGeometryToPositionsArray(g),
+          hierarchyArray: convertPolygonToHierarchyArray(g),
+        };
+      }
+      return {};
+    }, [g]);
+
+    const color = useMemo(
+      () => Color.fromCssColorString(stringColor ?? DEFAULT_SKETCH_COLOR),
+      [stringColor],
+    );
+
+    return (
+      <>
+        {positionsArray?.map((positions, index) => (
+          <PolylineEntity key={index} dynamic positions={positions} color={color} />
```

Comment:
This is just an idea or a memo or something like that...
You are using an Entity for polygon and polyline directly, but we already have Polygon and Polyline feature and appearance. I think we might want to customize the appearance for each sketched feature in the future.
However we need to implement each appearance to each feature. For example, we already have the appearance for Polygon feature, resource feature and sketch feature. Managing these three feature one by one is confusing us, so we should make a function which manage each appearance in one place later.